src/plugins/azgenerator/AzGeneratorFactory.ts

```diff
@@ -3,15 +3,23 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { Host } from "@azure-tools/autorest-extension-base";
+import { GenerationMode } from "../models";
 import { AzExtensionFullGenerator } from "./AzExtensionFullGenerator";
+import { AzExtensionIncrementalGenerator } from "./AzExtenSionIncrementalGenerator";
 import { AzGeneratorBase } from "./AzGeneratorBase";
-import { CodeModelAz } from "./CodeModelAz";
+import { CodeModelCliImpl } from "./CodeModelAzImpl";
 
 export class AzGeneratorFactory {
-    static async createAzGenerator(model: CodeModelAz, isDebugMode: boolean): Promise<AzGeneratorBase> {
+    static async createAzGenerator(host: Host, model: CodeModelCliImpl, isDebugMode: boolean): Promise<AzGeneratorBase> {
         await model.init();
         model.GenerateTestInit();
 
-        return new AzExtensionFullGenerator(model, isDebugMode);
+        if (model.codeModel.language['az']['generation-mode'] == GenerationMode.Full) {
```

Comment:
how about model.Extension_GenerationMode ?

Reply:
Yes, replaced 

---
In reply to: [472062452](https://github.com/Azure/autorest.az/pull/519#discussion_r472062452) [](ancestors = 472062452)